node/lib/util/status_util.js

```diff
@@ -465,8 +465,19 @@ exports.getRepoStatus = co.wrap(function *(repo, options) {
                                                     options.paths,
                                                     options.ignoreIndex,
                                                     options.showAllUntracked);
-        args.staged = status.staged;
-        args.workdir = status.workdir;
+        // if showMetaChanges is off, keep .gitmodules changes only
```

Comment:
Hm, is that really the right thing to do?  isn't a .gitmodules change a meta change?

Reply:
Also, this function is getting really long -- would it be possible to refactor somehow?

Reply:
let me see what I can do